### ISSUE TITLE
Static column support

### ISSUE THREAD

**Issue by anthonymejia** [Stratio/cassandra-lucene-index#70]
When do you expect to release support for static columns

**Comment by anthonymejia** [comment]
@adelapena any time frames on this?

**Comment by adelapena** [comment]
@anthonymejia we have plans to add support for static columns in 2.x, probably in not more than a month.

**Comment by sayeaud** [comment]
@anthonymejia -- just curious if there's any update on the ETA for this?

**Comment by anthonymejia** [comment]
@adelapena -- Curious too haha

**Comment by ealonsodb** [comment]
Hi @sayeaud  and @anthonymejia. 
We have been working on it and now in [branch-2.2.4](https://github.com/Stratio/cassandra-lucene-index/tree/branch-2.2.4) it returns static columns but it is not already posible to index them.
It (return static columns) will be included in next release
Thanks

**Comment by lucevers** [comment]
Upgrade to Apache Cassandra 2.2.5 ->  Is indexing possible for static columns in this release?

**Comment by adelapena** [comment]
No, it isn't. The last releases only allow indexing non-static columns in tables containing static column, but the static columns themselves can't be indexed. 

In future releases we could add the possibility of indexing only static columns, but there seems to not be an easy, performance wise acceptable, way of indexing both static and not-static columns in the same index. This is due to the differences between C* and Lucene data models.

**Comment by ddorian** [comment]
@adelapena Can't you just repeat the static column value on each document ? Since it will happen alot of times, the inverted index will mostly be compressed. Though there is an issue on update, since you have to update all the documents on that partition.

Or you can do something like elassandra http://doc.elassandra.io/en/latest/mapping.html#indexing-cassandra-static-columns

**Comment by adelapena** [comment]
Hi @ddorian ,

I think that updating all the documents in a partition when there is a change in one of their static columns is very risky. Performace-wise, it would be a very costly operation. We don't have a way to know if a static column write is an insert or an update, so we would need to read and reindex all the partition, which can contains millions of rows/documents.

I think I prefer the elassandra approach, indexing static columns in a separate document. Given the following table and index:

```
CREATE TABLE test(
pk int,
ck int,
sc int STATIC,
rc int,
 PRIMARY KEY (pk,ck)
);

CREATE CUSTOM INDEX test_idx ON test()
USING 'com.stratio.cassandra.lucene.Index'
 WITH OPTIONS = {
    'schema' : '{
       fields : {
          pk  : {type: "long"},
          ck  : {type: "long"},
          rc : {type: "long"},
          sc : {type: "long"}
       }
    }'
};
```
It would be relatively easy to search for full partitions:
```
SELECT * FROM test WHERE expr(test_idx, '{filter:
  {type: "match", field: "sc", value: 0}
}'); -- Returns all the rows in those partitions where sc=0
```
However, we could be interested in filtering by both static and non-static fields:
```
SELECT * FROM test WHERE expr(test_idx, '{filter: [
  {type: "match", field: "sc", value: 0},
  {type: "match", field: "rc", value: 0} ]
}');
```
This would be more complicated. We'd need to extract two Lucene queries from the user search, one for static columns, and other for non-static columns, and then [join them](https://lucene.apache.org/core/5_5_0/join/org/apache/lucene/search/join/package-summary.html) by the partition key field.

I think that this feature can be very interesting and we will start working on it as soon as we have time.
